### PR TITLE
Include X-Forwarded-Prefix in returnURL

### PR DESCRIPTION
**Pull request by andsens**
When using the PathPrefixStrip frontend rule in traefik X-Forwarded-Uri does not contain the true request URI, but the one where the path prefix has already been stripped.
Including the titular header in returnUrl() ensures that the user is returned to the correct URI when returning from the auth site.

See https://docs.traefik.io/basics/#path-matcher-usage-guidelines for more details. I'm not really sure what happens when you do a regexp rewrite though and how to handle that.

This solves my use-case though and works like a charm :-)

I also updated the tests.

### CODE REVIEW

**Comment by andsens** [comment]
Just saw [this one](https://docs.traefik.io/basics/#modifiers)

> ReplacePathRegex: ^/api/v2/(.*) /api/$1: Replaces the path with a regular expression and adds the old path to the X-Replaced-Path header. Separate the regular expression and the replacement by a space.

Man, one would think they could just add a `X-Request-URI` huh?

**Comment by thomseddon** [comment]
This is a great PR, thanks :)

Would you be willing to handle the `X-Replaced-Path` case too?

**Comment by andsens** [comment]
I would, not in this PR however. I'm sure it'll pop up as a challenge for me in the future, but I can't promise *when* I'll get around to it.

**Comment by andsens** [comment]
Welp, guess I ran into this sooner than expected. There you go, `X-Replaced-Path` is now also handled :-)

**Comment by thomseddon** [comment]
This will be merged in #49

File: forwardauth.go [hunk 1]
```diff
@@ -199,9 +199,16 @@ func (f *ForwardAuth) redirectBase(r *http.Request) string {
...
+	if replacedPath != "" {
+		// request-modifier and path-related matchers cannot be used at the same time
+		// so if X-Replaced-Path is set that means X-Forwarded-Prefix et al. are not
+		return fmt.Sprintf("%s%s", f.redirectBase(r), replacedPath)
+	}
+	prefix := r.Header.Get("X-Forwarded-Prefix")
 	path := r.Header.Get("X-Forwarded-Uri")
 
-	return fmt.Sprintf("%s%s", f.redirectBase(r), path)
+	return fmt.Sprintf("%s%s%s", f.redirectBase(r), prefix, path)
```
Comment (jongiddy) [inline forwardauth.go]: If Traefik is configured through Kubernetes ingress handlers, and the router is `/a/b/` and the path requested is `/a/b/`, then `traefik-forward-auth` will receive
```
X-Forwarded-Prefix: /a/b/
X-Forwarded-Uri: /
```

I found that setting `path` using this function works:
```
func GetUriPath(r *http.Request) string {
	prefix := r.Header.Get("X-Forwarded-Prefix")
	uri := r.Header.Get("X-Forwarded-Uri")
	p := path.Join(prefix, uri)
	if strings.HasSuffix(uri, "/") {
		p += "/"
	}
	return p
}
```

Comment (seuf) [reply forwardauth.go]: we whould not add another / at the end of the query if there is already one.

Comment (jongiddy) [reply forwardauth.go:211]: Under the circumstances that I have described, the `fmt.Sprintf` adds two slashes to the end.

With the code I've suggested, the `path.Join` always strips trailing slashes, so we add it back if it is in the original URL. It only leaves one trailing slash at the most.

**Pull request closed without merging**